### ISSUE TITLE
gemini create order not working

### ISSUE THREAD

**Issue by hzmbeyond80** [ccxt/ccxt#436]
i was able to query the balance successfully, which means the API key is working.  The API key has trading priority as well.   here is the code snippet 

```Python
gemini_balance = gemini.fetch_balance()
print 'gemini', gemini_balance['BTC']['total'], gemini_balance['USD']['total'] # success
time.sleep(2)
gemini.create_order('BTC/USD', 'limit', 'sell', 0.1, 6350) # got ccxt.base.errors.ExchangeNotAvailable
```

**Comment by kroitor** [comment]
@hzmbeyond80 can you post verbose output, plz?

**Comment by hzmbeyond80** [comment]
```
('https://api.gemini.com/v1/order/new', 'POST', 'https://api.gemini.com/v1/order/new', '\nRequest:', {'X-GEMINI-SIGNATURE': '9ec27cc961ba36ffcf77a1bddbd75bbe2240a47467b191f43f8fafbe7c364f152b073cc47744fa8bf5dc1548f6afe845', 'X-GEMINI-PAYLOAD': 'eyJub25jZSI6MTUwOTUyMDc0Niwic3ltYm9sIjoiYnRjdXNkIiwicmVxdWVzdCI6Ii92MS9vcmRlci9uZXciLCJjbGllbnRfb3JkZXJfaWQiOjE1MDk1MjA3NDYsImFtb3VudCI6IjAuMSIsInR5cGUiOiJleGNoYW5nZSBsaW1pdCIsInByaWNlIjoiNjM1MCIsInNpZGUiOiJzZWxsIn0=', 'Accept-Encoding': 'gzip, deflate', 'X-GEMINI-APIKEY': 'KuNk8fnPp8QwfHuu9ADM', 'Content-Type': 'text/plain', 'User-Agent': 'ccxt/1.9.307 (+https://github.com/ccxt-dev/ccxt) Python/2.7.10'}, None)
Traceback (most recent call last):
  File "/Users/zhiminhe/Dropbox/BitcoinPYVersions/BitCoinPYPython2.0/btc_action_jp_us.py", line 63, in <module>
    print gemini.create_order('BTC/USD', 'limit', 'sell', 0.1, 6350)
  File "/Users/zhiminhe/Dropbox/BitcoinPYVersions/BitCoinPYPython2.0/ccxt/gemini.py", line 162, in create_order
    response = self.privatePostOrderNew(self.extend(order, params))
  File "/Users/zhiminhe/Dropbox/BitcoinPYVersions/BitCoinPYPython2.0/ccxt/gemini.py", line 198, in request
    response = self.fetch2(path, api, method, params, headers, body)
  File "/Users/zhiminhe/Dropbox/BitcoinPYVersions/BitCoinPYPython2.0/ccxt/base/exchange.py", line 235, in fetch2
    return self.fetch(request['url'], request['method'], request['headers'], request['body'])
  File "/Users/zhiminhe/Dropbox/BitcoinPYVersions/BitCoinPYPython2.0/ccxt/base/exchange.py", line 288, in fetch
    self.handle_rest_errors(e, e.code, text, url, method)
  File "/Users/zhiminhe/Dropbox/BitcoinPYVersions/BitCoinPYPython2.0/ccxt/base/exchange.py", line 331, in handle_rest_errors
    self.raise_error(error, url, method, exception if exception else http_status_code, details)
  File "/Users/zhiminhe/Dropbox/BitcoinPYVersions/BitCoinPYPython2.0/ccxt/base/exchange.py", line 220, in raise_error
    details,
ccxt.base.errors.ExchangeNotAvailable: gemini POST https://api.gemini.com/v1/order/new 400 Bad Request  (possible reasons: invalid API keys, bad or old nonce, exchange is down or offline, on maintenance, DDoS protection, rate-limiting, )

Process finished with exit code 1
```

**Comment by kroitor** [comment]
@hzmbeyond80 thx! I'll take a look into it and will get back to you asap!

**Comment by hzmbeyond80** [comment]
*i found the cause.  all you need to do is changing the line marked with ####*

def create_order(self, symbol, type, side, amount, price=None, params={}):
    self.load_markets()
    if type == 'market':
        raise ExchangeError(self.id + ' allows limit orders only')
    order = {
        'client_order_id': str(self.nonce()),  #### add str()
        'symbol': self.market_id(symbol),
        'amount': str(amount),
        'price': str(price),
        'side': side,
        'type': 'exchange limit',  # gemini allows limit orders only
    }
    print order
    response = self.privatePostOrderNew(self.extend(order, params))
    return {
        'info': response,
        'id': response['order_id'],
    }

**Comment by kroitor** [comment]
@hzmbeyond80 awesome! Thanks for your help! I am going to upload the fix for it now!

**Comment by kroitor** [comment]
@hzmbeyond80 this was fixed in 1.9.329. Thx to you!
